pkg/media/pipeline.go

```diff
@@ -276,9 +276,35 @@ func (p *Pipeline) SendEOS(ctx context.Context) {
 	defer span.End()
 
 	p.closed.Once(func() {
-		logger.Debugw("sending EOS to pipeline")
-		p.input.Close()
-		p.sink.Close()
+		logger.Debugw("closing pipeline")
+
+		c := make(chan struct{})
+
+		go func() {
+			err := p.pipeline.BlockSetState(gst.StateNull)
+			if err != nil {
+				logger.Errorw("failed stopping pipeline", err)
+			}
+			p.loop.Quit()
```

Comment:
I think you'll want to quit the loop even on freeze

Reply:
Done

I also made a change to the WHIP handler to avoid blocking the teardown on a network read timeout.